stdlib/builtin/regexp.rbs

```diff
@@ -1,123 +1,359 @@
-  # `Regexp::new` to a regular expression with the same semantics as the
-  # original. (However, `Regexp#==` may not return true when comparing the
-  # two, as the source of the regular expression itself may differ, as the
-  # example shows). `Regexp#inspect` produces a generally more readable
-  # version of *rxp* .
-  #
-  # ```ruby
-  # r1 = /ab+c/ix           #=> /ab+c/ix
-  # s1 = r1.to_s            #=> "(?ix-m:ab+c)"
-  # r2 = Regexp.new(s1)     #=> /(?ix-m:ab+c)/
-  # r1 == r2                #=> false
-  # r1.source               #=> "ab+c"
-  # r2.source               #=> "(?ix-m:ab+c)"
-  # ```
+  # Returns a string containing the regular expression and its options (using the
+  # `(?opts:source)` notation. This string can be fed back in to Regexp::new to a
+  # regular expression with the same semantics as the original. (However,
+  # `Regexp#==` may not return true when comparing the two, as the source of the
+  # regular expression itself may differ, as the example shows).  Regexp#inspect
+  # produces a generally more readable version of *rxp*.
+  #
+  #     r1 = /ab+c/ix           #=> /ab+c/ix
+  #     s1 = r1.to_s            #=> "(?ix-m:ab+c)"
+  #     r2 = Regexp.new(s1)     #=> /(?ix-m:ab+c)/
+  #     r1 == r2                #=> false
+  #     r1.source               #=> "ab+c"
+  #     r2.source               #=> "(?ix-m:ab+c)"
+  #
   def to_s: () -> String
 
+  # Match---Matches *rxp* against the contents of `$_`. Equivalent to *`rxp* =~
+  # $_`.
+  #
+  #     $_ = "input data"
+  #     ~ /at/   #=> 7
+  #
   def ~: () -> Integer?
 
-  def self.compile: (String arg0, ?untyped options, ?String kcode) -> Regexp
-                  | (Regexp arg0) -> Regexp
-
-  def self.quote: (String | Symbol arg0) -> String
+  private
 
-  def self.union: () -> Regexp
-                | (String | Regexp arg0, *String | Regexp args) -> Regexp
-                | (::Array[String | Regexp]) -> Regexp
+  # Constructs a new regular expression from `pattern`, which can be either a
+  # String or a Regexp (in which case that regexp's options are propagated), and
+  # new options may not be specified (a change as of Ruby 1.8).
+  #
+  # If `options` is an Integer, it should be one or more of the constants
+  # Regexp::EXTENDED, Regexp::IGNORECASE, and Regexp::MULTILINE, *or*-ed together.
+  #  Otherwise, if `options` is not `nil` or `false`, the regexp will be case
+  # insensitive.
+  #
+  #     r1 = Regexp.new('^a-z+:\\s+\w+') #=> /^a-z+:\s+\w+/
+  #     r2 = Regexp.new('cat', true)     #=> /cat/i
+  #     r3 = Regexp.new(r2)              #=> /cat/i
+  #     r4 = Regexp.new('dog', Regexp::EXTENDED | Regexp::IGNORECASE) #=> /dog/ix
+  #
+  incompatible def initialize: (String arg0, ?untyped options, ?String kcode) -> Object
+                             | (Regexp arg0) -> void
 
-  def eql?: (untyped other) -> bool
+  def initialize_copy: (self object) -> self
 end
 
 Regexp::EXTENDED: Integer
```

Comment:
It looks like RDoc has entries for these constants. Might be a bug of `bin/annotate-with-rdoc`...

Could you add RDoc comments for these? The following is the comment.

```
# see Regexp.options and Regexp.new
# 
```

(It may be essentially useless, but I want to have RDoc content in RBS too.)

Reply:
Some parts of the document including `# see Regexp.options and Regexp.new` were lost because I applied `rbs prototype runtime --merge` after `bin/annotate-with-rdoc`...🙏
Then I appilied `bin/annotate-with-rdoc` again and fix them.